### ISSUE TITLE
OpenTable Reservation block calls are inconsistent

### ISSUE THREAD

**Issue by paulopmt1** [Automattic/jetpack#39410]
### Quick summary

Inconsistency return of the `/restaurant-search` endpoint from OpenTable is breaking our e2e tests.

### Steps to reproduce

1. Create a new `OpenTable Reservation` block. 
2. Paste this restaurant: `Miku Restaurant - Vancouver`
3. Erase and paste again, and you'll see the results are not consistent:

<img width="1506" alt="image" src="https://github.com/user-attachments/assets/99ea81d2-3ee6-439b-a5e5-0adfc35b3f1c">

### What you expected to happen

To always have a 200 return

### What actually happened

1/3 of Open table search calls return error 500

### Impact

One

### Available workarounds?

No and the platform is unusable

**Comment by paulopmt1** [comment]
After fixing our internal e2e sites, I contacted OpenTable to let them know about the API inconsistency 8503909-zd-a8c

**Comment by paulopmt1** [comment]
Sometimes, the failure frequency is even higher than 1/3: 

<img width="852" alt="image" src="https://github.com/user-attachments/assets/cca3fe9b-7c0a-4569-93ed-f7ec2dc8291a">

**Comment by jeherve** [comment]
Moving this to the Jetpack repo, where the block is being developed.

**Comment by jeherve** [comment]
The calls now seem to fail almost all the time.

I contacted the OpenTable team about this in 8741233-zen.

**Comment by jeherve** [comment]
@monsieur-z Until that issue is fixed with the endpoint, do you think we could catch the failed API request response, and display an error in the block, outputting something like "OpenTable does not find this restaurant right now. Please try again later"?

**Comment by jeherve** [comment]
> I contacted the OpenTable team about this in 8741233-zd-a8c.

To provide an update on the conversation still happening with the OpenTable team:


- In the block, we currently rely on hitting `https://www.opentable.com/widget/reservation/restaurant-search?pageSize=20&query=` to get a list of restaurant IDs that match a specific keyword.
- We then use that restaurant ID to build the proper embed.
- That URL above no longer works, it returns 500 errors.
- OpenTable has confirmed that this URL is not meant to be used to search for restaurants. It seems we were consequently using an undocumented solution, that OpenTable no longer supports.
- I've asked OpenTable for a supported alternative.

**Comment by kraftbj** [comment]
At this point, OpenTable does not provide any alternatives to the search functionality. I've asked again today, but I'm afraid we're likely needing to strip down block functionality only to accept the widget URL that restaurant owners get from the OpenTable system.

I don't know if that is valuable on its own.

**Comment by donnapep** [comment]
Hi @paulopmt1. Just checking in on this high priority issue to see if there's anything new to report?

**Comment by paulopmt1** [comment]
Hey @donnapep, thanks for checking in!

The next step is a product decision about this block, as @kraftbj noticed: "we're likely needing to strip down block functionality only to accept the widget URL that restaurant owners get from the OpenTable system."

**Comment by pablinos** [comment]
> OpenTable has confirmed that this URL is not meant to be used to search for restaurants. It seems we were consequently using an undocumented solution, that OpenTable no longer supports.

Yes, it wasn't strictly documented. They had a widget builder on their site, and we rebuilt it in the block. I don't know if they still have one of those?

**Comment by getdave** [comment]
I tested this today and the search endpoint now seems to _always_ return `500`

**Comment by scruffian** [comment]
I think to unblock this we need an OpenTable restaurant account with access to their widget builder, or a contact at OpenTable.

**Comment by draganescu** [comment]
No we need to as @kraftbj noticed: "we're likely needing to strip down block functionality only to accept the widget URL that restaurant owners get from the OpenTable system."

**Comment by monsieur-z** [comment]
@draganescu, I'm trying to close down some of the block-related issues. Is this PR ready to ship? Anything I can help with?

**Comment by draganescu** [comment]
@monsieur-z yes the PR seems to have gotten some updates and is approved. Will merge as soon as I rebase it.

**Comment by monsieur-z** [comment]
Thanks @draganescu!
